src/index.ts

```diff
@@ -771,20 +775,93 @@ class Spanner extends GrpcService {
         callback(err);
         return;
       }
-      const clientName = config.client;
+      this.getGaxClient_(config, (err, gaxClient) => {
+        if (err) {
+          callback(err);
+          return;
+        }
+        let reqOpts = extend(true, {}, config.reqOpts);
+        reqOpts = replaceProjectIdToken(reqOpts, projectId!);
+        const requestFn = gaxClient[config.method].bind(
+          gaxClient,
+          reqOpts,
+          config.gaxOpts
+        );
+        callback(null, requestFn);
+      });
+    });
+  }
+
+  /**
+   * get GAX client. This will retrive the end point uris for specific instance and cache the client accordingly.
+   *
+   * @private
+   *
+   * @param {string} projectId Request config
+   * @param {object} config Request config
+   * @param {function} callback Callback function
+   */
+  getGaxClient_(config, callback) {
+    let clientName = config.client;
+    if (
+      clientName !== 'SpannerClient' ||
+      !this.options.enableResourceBasedRouting
+    ) {
       if (!this.clients_.has(clientName)) {
-        this.clients_.set(clientName, new gapic.v1[clientName](this.options));
+        this.setSpannerClient(clientName, config, this.options);
       }
       const gaxClient = this.clients_.get(clientName)!;
-      let reqOpts = extend(true, {}, config.reqOpts);
-      reqOpts = replaceProjectIdToken(reqOpts, projectId!);
-      const requestFn = gaxClient[config.method].bind(
-        gaxClient,
-        reqOpts,
-        config.gaxOpts
-      );
-      callback(null, requestFn);
-    });
+      callback(null, gaxClient);
+      return;
+    }
+
+    if (!config.instanceId) {
+      const error = new Error('instanceId is required.');
+      callback(error);
+      return;
+    }
+    const instanceId = config.instanceId;
+    clientName = `${clientName}-${instanceId}`;
+
+    if (!this.clients_.has(clientName)) {
+      this.instances_
+        .get(instanceId)!
+        .getInstanceEndPointUris((err, endPointUris) => {
```

Comment:
Please add a comment above this line (to remember to change before merging)
That once #760 PR is merged this will get replaced with 
```ts
getMetadata({fieldNames: 'displayName'}, (err, metadata, apiResponse)) => {...}
```

Reply:
`endPointUris` => `endpointUris` here and down below please

Reply:
Done!!